src/libraries/System.Net.Http/tests/FunctionalTests/HttpConnectionKeyTest.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public class HttpConnectionKeyTest
+    {
+        public static IEnumerable<object[]> KeyComponents()
+        {
+            yield return new object[] { "Https", "localhost", 80, "localhost-ssl", new Uri("http://localhost"), "domain1/userA", false};
+            yield return new object[] { "Http", "localhost1", 80, "localhost-ssl", new Uri("http://localhost"), "domain1/userA", false };
+            yield return new object[] { "Http", "localhost", 81, "localhost-ssl", new Uri("http://localhost"), "domain1/userA", false };
+            yield return new object[] { "Http", "localhost", 80, "localhost-ssl1", new Uri("http://localhost"), "domain1/userA", false };
+            yield return new object[] { "Http", "localhost", 80, "localhost-ssl", new Uri("http://localhost1"), "domain1/userA", false };
+            yield return new object[] { "Http", "localhost", 80, "localhost-ssl", new Uri("http://localhost"), "domain1/userB", false };
+            yield return new object[] { "Http", "localhost", 80, "localhost-ssl", new Uri("http://localhost"), "domain1/userA", true };
+        }
+
+        [Theory, MemberData(nameof(KeyComponents))]
+        public void Equals_DifferentParameters_ReturnsTrueIfAllEqual(string kindString, string host, int port, string sslHostName, Uri proxyUri, string identity, bool expected)
```

Comment:
>Since this is all dependent on product code, can it be made a unit test instead?

I agree it is preferred to make this a pure unit test as long as the types (HttpConnectionKind, etc.) are easily consumable when building the test assembly. But if not, it's fine to use a little reflection.

@alnikola The pure System.Net.Http/tests/UnitTests are tests where we build the product sources directly into the test assembly. That gives us access to 'internal' classes and methods. This is a technique we don't use in a lot of places in the repo. But it's useful for things like this.